public/app/features/dashboard-scene/sharing/components/ImagePreview.tsx

```diff
@@ -0,0 +1,119 @@
+import { css } from '@emotion/css';
+import { useMemo, useEffect } from 'react';
+import { useMeasure } from 'react-use';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { t } from '@grafana/i18n';
+import { Alert, LoadingBar, Text, useStyles2 } from '@grafana/ui';
+
+type ErrorState = {
+  message: string;
+  title: string;
+  code?: string;
+} | null;
+
+interface ImagePreviewProps {
+  imageBlob: Blob | null;
+  isLoading: boolean;
+  error: ErrorState;
+  testId?: string;
+  title?: string;
+  showLoading?: boolean;
+}
+
+export function ImagePreview({
+  imageBlob,
+  isLoading,
+  error,
+  testId = selectors.components.ExportImage.preview.container,
+  title,
+  showLoading = true,
+}: ImagePreviewProps) {
+  const styles = useStyles2(getStyles);
+  const [ref, { width: measuredWidth }] = useMeasure<HTMLDivElement>();
+
+  // Memoize and clean up the object URL for the image
+  const imageUrl = useMemo(() => {
+    if (!imageBlob) {
+      return undefined;
+    }
+    const url = URL.createObjectURL(imageBlob);
+    return url;
+  }, [imageBlob]);
+
+  useEffect(() => {
+    return () => {
+      if (imageUrl) {
+        URL.revokeObjectURL(imageUrl);
+      }
+    };
+  }, [imageUrl]);
+
+  return (
+    <div className={styles.previewContainer} ref={ref} data-testid={testId}>
+      {showLoading && isLoading && (
+        <div className={styles.loadingBarContainer} data-testid={selectors.components.ExportImage.preview.loading}>
+          <LoadingBar width={measuredWidth} />
+          {title && (
```

Comment:
there's always a title

Reply:
this is an optional prop so there may be a case where there isn't a title provided 😬

Reply:
see https://github.com/grafana/grafana/pull/104207/files/fa940936bfa55dd61656ffbfdbd22b2d0d1f4488#r2208026255